### ISSUE TITLE
Add a flag to ignore dependency overrides in pubspec

### ISSUE THREAD

**Issue by alorenzen** [dart-lang/pub#1867]
In angular, we use a "mono_repo" approach, where we keep multiple related packages in the same repo, and test/publish them at once.

In order to support local development and testing on travis, we use `dependency_overrides` in our pubspec to point to the local packages and not the published versions. The only time we don't want this behavior is when we are actually publishing new packages.

Right now, we have to manually comment out our `dependency_overrides` sections in every `pubspec.yaml` before we can run `pub upgrade` and then `pub publish`. This manual step is a burden, and something blocking us from fully automating our publish process.

Can we add a `--ignore-overrides` flag to pub (specifically the `get`, `upgrade`, and `publish` commands) to remove the need for this manual removal step?

**Comment by nex3** [comment]
I'm not sure what your local dependencies have to do with publishing...

**Comment by jakemac53** [comment]
It allows you to test the package against the actual published versions of all dependencies as opposed to the local overrides.

**Comment by alorenzen** [comment]
So `angular_forms` depends on `angular`, which depends on `angular_compiler`, which depends on `angular_ast`.

We have a dependency_override in `angular_forms` for `angular`, `angular_compiler`, and `angular_ast`.  https://github.com/dart-lang/angular/blob/master/angular_forms/pubspec.yaml#L22

*After* we publish `angular`, but *before* we publish `angular_forms`, we need to remove the dependency_overrides and verify that everything is working with the live versions. (I.e. we didn't forget to bump the version for `angular`).

**Comment by nex3** [comment]
All right, sounds good to me. I've also been meaning to add a `--no-dev` flag for a while, which is similar.

**Comment by jonasfj** [comment]
I don't think a `--no-dev` flag will work since you won't get dev-dependencies needed for tests.

But this might be worth taking into the discussion in #2036.

Note. It's also tempting to use `mono_repo` to rewrite pubspecs to comment in/out the `dependency_overrides`.

**Comment by lemz90** [comment]
Docker Compose uses an override file to solve this - [docker-compose.override.yaml](https://docs.docker.com/compose/extends/)

If the override file exists, the content inside is used, otherwise it just uses the docker-compose.yaml like normal.

The override file can then be ignored in version control. I might work on a PR for this. Thoughts?

**Comment by lemz90** [comment]
@nex3 I have some working code for my proposed solution in [my comment](https://github.com/dart-lang/pub/issues/1867#issuecomment-499182381) above.

I was reading the dartlang contribution guidelines and it seems like you do not take PRs.

Not sure if I should go through with the Rietveld review since I was really just going to make a PR as a conversation starter (the code is nothing special and probably not up to dart team par). Anyway, I think it would be nice to have something like this functionality.

If you do not want to add something like this, let me know so I can start work on a command line utility or something to do this.

**Comment by nex3** [comment]
I'm no longer on the Dart team, but when I was, pub took normal GitHub pull requests.

**Comment by jonasfj** [comment]
@lemz90, we do accept PRs :)

Sorry, I was a little slow... Let's discuss this on the PR you filed... Regardless of whether we have an override file or not, it might be desirable to have a flag for disabling it, see for example: https://github.com/dart-lang/pub-dev/issues/2355

**Comment by jonasfj** [comment]
@lemz90, we do accept PRs :)

Sorry, I was a little slow... Let's discuss this on the PR you filed... Regardless of whether we have an override file or not, it might be desirable to have a flag for disabling it, see for example: https://github.com/dart-lang/pub-dev/issues/2355

**Comment by lemz90** [comment]
@alorenzen @jakemac53 Hey guys, we have more discussion past just adding `--ignore-overrides` in a new issue #2161

Please post any ideas or input :)

**Comment by natebosch** [comment]
I think we can consider this closed by https://github.com/dart-lang/pub/pull/3215
